datasetinsights/io/downloader/http_downloader.py

```diff
@@ -51,3 +62,74 @@ def download(self, source_uri, output, checksum_file=None, **kwargs):
                 logger.info("Checksum mismatch. Deleting the downloaded file.")
                 os.remove(dataset_path)
                 raise e
+
+
+def download_dataset_from_http_url(source_uri, output):
+    """ Downloads dataset from HTTP(S) URL and detects name of the file to be
+        downloaded.
+
+        Args
+            source_uri (str): This is the downloader-uri that indicates where
+                              the dataset should be downloaded from.
+
+            output (str): This is the path to the directory where the
+                          download will store the dataset.
+    """
+    output = Path(output)
+
+    adapter = TimeoutHTTPAdapter(
+        timeout=DEFAULT_TIMEOUT, max_retries=Retry(total=DEFAULT_MAX_RETRIES)
+    )
+    with requests.Session() as http:
+        http.mount("https://", adapter)
+        try:
+            response = http.get(source_uri)
+            response.raise_for_status()
+        except requests.exceptions.RequestException as ex:
+            logger.error(ex)
+            err_msg = (
+                f"The request download from {source_uri} -> {output} can't "
+                f"be completed."
+            )
+            raise DownloadError(err_msg)
+        else:
+            output.parent.mkdir(parents=True, exist_ok=True)
+
+            file_name = get_filename_from_response(response)
+            if file_name is None:
+                file_name = get_file_name_from_uri(source_uri)
+
+            dataset_path = output / file_name
+
+            with open(dataset_path, "wb") as f:
+                f.write(response.content)
+    return dataset_path
+
+
+def get_filename_from_response(response):
+    """ Gets filename from requests response object
+
+        Args:
+            response: requests.Response() object that contains the server's
+            response to the HTTP request.
+
+        Returns:
+            filename (str): Name of the file to be downloaded
+    """
+    cd = response.headers.get("content-disposition")
+    if not cd:
+        return None
+    file_name = re.findall("filename=(.+)", cd)
+    if len(file_name) == 0:
+        return None
+    return file_name[0]
+
+
+def get_file_name_from_uri(uri):
```

Comment:
URL is not always reliable to parse filenames. It might have http redirect which hides details of files it contains. It should be sufficient to get filename from http responses.

Reply:
Not every time HTTP response contains content-disposition header.